scripts/test/TOFTOF/TOFTOFGUITest.py

```diff
@@ -0,0 +1,133 @@
+from __future__ import (absolute_import, division, print_function)
+
+from reduction_gui.reduction.toftof.toftof_reduction import TOFTOFScriptElement, OptionalFloat
+from reduction_gui.widgets.toftof.toftof_setup import TOFTOFSetupWidget
+import mantid
+from PyQt4 import QtGui
+
+import unittest
+try:
+    from unittest import mock
+except ImportError:
+    import mock
+
+
+class TOFTOFScriptElementTest(unittest.TestCase):
+    @classmethod
+    def setUpClass(cls):
+        cls.app = QtGui.QApplication([])
+
+    def setUp(self):
+        self.settingsMock = mock.Mock()
+        facility_name = "name of facility"
+        self.settingsMock.facility_name = facility_name
+        instrument_name = "name of instrument"
+        self.settingsMock.instrument_name = instrument_name
+        self.setupWidget = TOFTOFSetupWidget(self.settingsMock)
+
+    def tearDown(self):
+        self.settingsMock = None
+        self.setupWidget = None
+        #pass
+        #self.scriptElement = None
+        #done
+
+    #def test_that_init_works_correctly(self):
+
+    def test_that_preserves_data(self):
+        scriptElement = TOFTOFScriptElement()
+        scriptElement.reset()
+        scriptElement.facility_name   = 'nonsense'
+        scriptElement.instrument_name = 'The 5th nonSense'
+
+        # prefix of (some) workspace names
+        scriptElement.prefix   = 'ws'
+
+        # data files are here
+        scriptElement.dataDir  = ''
+
+        # vanadium runs & comment
+        scriptElement.vanRuns  = 'TOFTOFTestdata.nxs'
+        scriptElement.vanCmnt  = 'vanadium comment'
+        scriptElement.vanTemp  = OptionalFloat(None)
+
+        # empty can runs, comment, and factor
+        scriptElement.ecRuns   = 'TOFTOFTestdata.nxs'
+        scriptElement.ecTemp   = OptionalFloat(21.0)
+        scriptElement.ecFactor = 0.9
+
+        # data runs: [(runs,comment, temperature), ...]
+        scriptElement.dataRuns = [
+            [u'TOFTOFTestdata.nxs', u'H2O 21C', OptionalFloat(None)],
+            [u'TOFTOFTestdata.nxs', u'H2O 34C', OptionalFloat(34.0)]
```

Comment:
Just a note: my understanding is that the `u` prefix for strings is a syntax error in early Python 3 versions (3.0, 3.2?). I hope when Mantid finally moves to Python 3 everyone will be running a later version...